ludwig/data/dataset_synthesizer.py

```diff
@@ -111,7 +113,12 @@ def return_none(feature):
 
 
 def assign_vocab(feature):
-    feature["idx2str"] = build_vocab(feature.get("vocab_size", 10))
+    if DECODER in feature:
+        feature["idx2str"] = build_vocab(feature[DECODER].get("vocab_size", 10))
+    elif ENCODER in feature:
+        feature["idx2str"] = build_vocab(feature[ENCODER].get("vocab_size", 10))
+    else:
+        feature["idx2str"] = build_vocab(10)
```

Comment:
this parameter is a parameter of the config for synthesizer. synthesizer configs shouldn't need encoder and decoder i believe

Reply:
So in this instance, the `vocab_size` parameter has moved into the encoder or the decoder depending on if the feature is an input or output feature. So this logic is basically allowing the `assign_vocab` function to grab it from the respective nested dictionary since some features have a `decoder` key but not an `encoder` key and vice versa.